### PR TITLE
Adding ELFHash Algorithm

### PR DESCRIPTION
**Pull request by micaelalex**
Adding a new Hash Algorithm.

### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: hashes/elf.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+def elf_hash(data: str) -> int:
+    """
+    Implementation of ElfHash Algorithm, a variant of PJW hash function.
+
+    Returns:
+        [int] -- [32 bit binary int]
+    >>> elf_hash('lorem ipsum')
+    253956621
+    """
+    hash = x = i = 0
```
Comment (CaedenPH) [inline hashes/elf.py:10]:
```suggestion
    hash = x = 0
```

File: hashes/elf.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    for i in range(len(data)):
+        hash = (hash << 4) + ord(data[i])
```
Comment (CaedenPH) [inline hashes/elf.py:12]:
```suggestion
    for letter in data:
        hash = (hash << 4) + ord(letter)
```

**Review by CaedenPH — APPROVED** [review]
@cclauss Looks good

**Review by cclauss — APPROVED** [review]
RSLGTM

**Pull request merged**